main-site/lib/types.ts

```diff
@@ -1,3 +1,22 @@
+import { Dispatch, SetStateAction } from 'react';
+
+export interface FaqPropsArray {
+  id: number;
+  question: string;
+  answer: string;
+}
+
+export interface FaqData {
+  data: Array<FaqPropsArray>;
+}
+
+export interface FaqProps {
+  id: number;
+  question: string;
+  answer: string;
+  selectedQuestion: number;
+  setSelectedQuestion: Dispatch<SetStateAction<number>>;
```

Comment:
Change this to   setSelectedQuestion: () => void;

Reply:
U might need to pass in something for the () like   setSelectedQuestion: (selected: SelectedQuestion) => void; where SelectedQuestion is the type (I think u have to create this also in types.ts)